### PR TITLE
EMR Serverless Create/Stop/Delete Application deferrable mode

### PR DESCRIPTION
**Pull request by syedahsn**
This PR adds deferrable mode to `EmrServerlessCreateApplication`, `EmrServerlessStopApplication`, and `EmrServerlessDeleteApplication`. This required a bit of change in the hook code, but hopefully its clear why I did that. Tests are included

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/emr.py [hunk 1]
```diff
@@ -1368,10 +1469,17 @@ class EmrServerlessDeleteApplicationOperator(EmrServerlessStopApplicationOperato
...
+        step of first,the application to be stopped, and then deleted. Defaults to 25 minutes.
+    :param waiter_check_interval_seconds: (deprecated) Number of seconds between polling the state
+        of the application. Defaults to 60 seconds.
+    :waiter_max_attempts: Number of times the waiter should poll the application to check the state.
+        Default is 25.
+    :param waiter_delay: Number of seconds between polling the state of the application.
+        Default is 60 seconds.
+    :param deferrable: If True, the operator will wait asynchronously for application to be deleted.
+        This implies waiting for completion. This mode requires aiobotocore module to be installed.
+        (default: False)
```
Comment (pragnareddye) [inline airflow/providers/amazon/aws/operators/emr.py]: Just keeping it consistent with other options
```suggestion
        Defaults to False
```

File: airflow/providers/amazon/aws/operators/emr.py [hunk 2]
```diff
@@ -1368,10 +1469,17 @@ class EmrServerlessDeleteApplicationOperator(EmrServerlessStopApplicationOperato
...
-    :param waiter_countdown: Total amount of time, in seconds, the operator will wait for each step of first,
-        the application to be stopped, and then deleted. Defaults to 25 minutes.
-    :param waiter_check_interval_seconds: Number of seconds between polling the state of the application.
-        Defaults to 60 seconds.
+    :param waiter_countdown: (deprecated) Total amount of time, in seconds, the operator will wait for each
+        step of first,the application to be stopped, and then deleted. Defaults to 25 minutes.
+    :param waiter_check_interval_seconds: (deprecated) Number of seconds between polling the state
+        of the application. Defaults to 60 seconds.
+    :waiter_max_attempts: Number of times the waiter should poll the application to check the state.
+        Default is 25.
```
Comment (pragnareddye) [inline airflow/providers/amazon/aws/operators/emr.py]:
```suggestion
        Defaults to 25.
```

File: airflow/providers/amazon/aws/operators/emr.py [hunk 3]
```diff
@@ -1368,10 +1469,17 @@ class EmrServerlessDeleteApplicationOperator(EmrServerlessStopApplicationOperato
...
+    :param waiter_delay: Number of seconds between polling the state of the application.
+        Default is 60 seconds.
```
Comment (pragnareddye) [inline airflow/providers/amazon/aws/operators/emr.py]:
```suggestion
        Defaults to 60 seconds.
```

File: airflow/providers/amazon/aws/hooks/emr.py [hunk 4]
```diff
@@ -258,7 +258,8 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 
     def cancel_running_jobs(self, application_id: str, waiter_config: dict = {}):
         """
-        List all jobs in an intermediate state, cancel them, then wait for those jobs to reach terminal state.
+        List all jobs in an intermediate state, cancel them, then returns the number of
+        cancelled jobs.
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/emr.py]:
```suggestion
        Cancel jobs in an intermediate state, and return the number of cancelled jobs.
```

We’re working on enforcing this docstring format soon.

File: airflow/providers/amazon/aws/hooks/emr.py [hunk 5]
```diff
@@ -284,13 +284,7 @@ def cancel_running_jobs(self, application_id: str, waiter_config: dict = {}):
...
                 for job_id in job_ids:
                     self.conn.cancel_job_run(applicationId=application_id, jobRunId=job_id)
-        if count > 0:
-            self.log.info("now waiting for the %s cancelled job(s) to terminate", count)
-            self.get_waiter("no_job_running").wait(
-                applicationId=application_id,
-                states=list(self.JOB_INTERMEDIATE_STATES.union({"CANCELLING"})),
-                WaiterConfig=waiter_config,
-            )
+        return count
```
Comment (vandonr-amz) [inline airflow/providers/amazon/aws/hooks/emr.py:301]: isn't that a breaking change ?
Also, if you were to go with it, you could remove the waiter_config param.

File: airflow/providers/amazon/aws/operators/emr.py [hunk 6]
```diff
@@ -1025,11 +1031,12 @@ def __init__(
         self.aws_conn_id = aws_conn_id
         self.release_label = release_label
         self.job_type = job_type
-        self.wait_for_completion = wait_for_completion
+        self.wait_for_completion = False if deferrable else wait_for_completion
```
Comment (vandonr-amz) [inline airflow/providers/amazon/aws/operators/emr.py:1037]: this is not really needed, is it ?
I was actually wondering with the new default if it'd make sense to defer only if `deferrable && wait_for_completion` ?

File: airflow/providers/amazon/aws/operators/emr.py [hunk 7]
```diff
@@ -1079,6 +1098,29 @@ def execute(self, context: Context) -> str | None:
             )
         return application_id
 
+    def start_application_deferred(self, context, event=None):
+        if event["status"] == "success":
```
Comment (vandonr-amz) [inline airflow/providers/amazon/aws/operators/emr.py:1105]: you could flip this if to reduce indentation

**Pull request closed without merging**